proxy/envoy/config.go

```diff
@@ -266,18 +266,23 @@ func buildRoutes(context *ProxyContext) (HTTPRouteConfigs, TCPRouteConfigs) {
 // buildOutboundRoutes creates route configs indexed by ports for the traffic outbound
 // from the proxy instance
 func buildOutboundRoutes(instances []*model.ServiceInstance, services []*model.Service,
-	config *model.IstioRegistry, mesh *MeshConfig) (HTTPRouteConfigs, TCPRouteConfigs) {
+	context *ProxyContext) (HTTPRouteConfigs, TCPRouteConfigs) {
 	// used for shortcut domain names for outbound hostnames
 	suffix := sharedInstanceHost(instances)
 	httpConfigs := make(HTTPRouteConfigs)
 	tcpConfigs := make(TCPRouteConfigs)
 
 	// get all the route rules applicable to the instances
-	rules := config.RouteRulesBySource("", instances)
+	rules := context.Config.RouteRulesBySource("", instances)
 
 	// outbound connections/requests are redirected to service ports; we create a
 	// map for each service port to define filters
 	for _, service := range services {
+		sslContext, err := buildSSLContext(service.Hostname, context)
+		if err != nil {
+			glog.Errorf("Error generating SSL context for service: %s. Skip.", service.Hostname)
```

Comment:
Again, we cannot have errors in the controllers. You need to print a warning and deal with it somehow. Catastrophic errors are not acceptable. And log errors are too opaque for the operator.